### PR TITLE
Doc: Update the DatetimeIndex.strftime docstring

### PR DESCRIPTION
**Pull request by Tushar987**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [ ] PR title is "DOC: update the <your-function-or-method> docstring"
- [ ] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [ ] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [ ] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
################## Docstring (pandas.DatetimeIndex.strftime)  ##################
################################################################################

Convert to specified date_format.

Return an array of formatted strings specified by date_format, which
supports the same string format as the python standard library. Details
of the string format can be found in `python string format doc <https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior>`__

Parameters
----------
date_format : str
    Date format string (e.g. "%Y-%m-%d").

Returns
-------
numpy.ndarray
    n-dimensional array of formatted strings

See Also
--------
DatetimeIndex.normalize : Return DatetimeIndex with times to midnight.
DatetimeIndex.round : Round the DatetimeIndex to the specified freq.
DatetimeIndex.floor : Floor the DatetimeIndex to the specified freq.

Examples
--------
>>> data = ['2015-05-01 18:47:05.060000','2014-05-01 18:47:05.110000']
>>> df = pd.DataFrame(data, columns=['date'])
>>> df.date = pd.to_datetime(df.date)
>>> df.date[1]
Timestamp('2014-05-01 18:47:05.110000')
>>> df.date[1].strftime('%d-%m-%Y')
'01-05-2014'
>>> df.date[1].strftime('%B %d, %Y, %r')
'May 01, 2014, 06:47:05 PM'

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DatetimeIndex.strftime" correct. :)
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

Checklist for other PRs (remove this part if you are doing a PR for the pandas documentation sprint):

### CODE REVIEW

File: pandas/core/indexes/datetimelike.py [hunk 1]
```diff
@@ -61,18 +61,39 @@ def strftime(self, date_format):
         return np.asarray(self.format(date_format=date_format),
                           dtype=compat.text_type)
     strftime.__doc__ = """
+    Convert to specified date_format.
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/datetimelike.py]: Maybe mention it is converted to a string?

File: pandas/core/indexes/datetimelike.py [hunk 2]
```diff
@@ -61,18 +61,39 @@ def strftime(self, date_format):
...
     -------
-    ndarray of formatted strings
+    numpy.ndarray
+        n-dimensional array of formatted strings
+
+    See Also
+    --------
+    DatetimeIndex.normalize : Return DatetimeIndex with times to midnight.
+    DatetimeIndex.round : Round the DatetimeIndex to the specified freq.
+    DatetimeIndex.floor : Floor the DatetimeIndex to the specified freq.
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/datetimelike.py]: I would also link to the similar conversion but other way around: to_datetime

File: pandas/core/indexes/datetimelike.py [hunk 3]
```diff
@@ -61,18 +61,39 @@ def strftime(self, date_format):
...
+    --------
+    >>> data = ['2015-05-01 18:47:05.060000','2014-05-01 18:47:05.110000']
+    >>> df = pd.DataFrame(data, columns=['date'])
+    >>> df.date = pd.to_datetime(df.date)
+    >>> df.date[1]
+    Timestamp('2014-05-01 18:47:05.110000')
+    >>> df.date[1].strftime('%d-%m-%Y')
+    '01-05-2014'
+    >>> df.date[1].strftime('%B %d, %Y, %r')
+    'May 01, 2014, 06:47:05 PM'
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/datetimelike.py]: The docstring is from DatetimeIndex. Can you update the example to use that? (so eg create an example dataframe with DatetimeIndex with eg `pd.date_range`)

File: pandas/core/indexes/datetimelike.py [hunk 4]
```diff
@@ -61,18 +61,39 @@ def strftime(self, date_format):
...
     ----------
     date_format : str
-        date format string (e.g. "%Y-%m-%d")
+        Date format string (e.g. "%Y-%m-%d").
 
     Returns
     -------
-    ndarray of formatted strings
+    numpy.ndarray
+        n-dimensional array of formatted strings
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/datetimelike.py]: I would just say "NumPy array" instead of "n-dimensional", as it is always 1D

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR! Added some comments

File: pandas/core/indexes/datetimelike.py (hunk 4 above)
Comment (jreback) [reply pandas/core/indexes/datetimelike.py]: @jorisvandenbossche separate issue, this should actually return an ``Index`` here.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @Tushar987

**Pull request merged**